flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/fetcher/SplitFetcherManagerTest.java

```diff
@@ -71,7 +71,7 @@ public void testCloseFetcherWithException() throws Exception {
                 .hasRootCauseMessage("Artificial exception on closing the split reader.");
     }
 
-    @Test(timeout = 30000)
+    @Test
```

Comment:
Sorry @becketqin , I don't why adding timeout to `Long.MAX_VALUE` will work when there are tasks not terminated. I did a test and confirmed even if there are tasks not terminated, the test still can pass, with or without the `timeout = Long.MAX_VALUE`. 

To catch the error when there are tasks failing to be closed, I change the `close` method to return `boolean`, so that we have a way to know if tasks are completely closed in time, otherwise, fail the test. Please take a look again. Thanks.

Reply:
>  I did a test and confirmed even if there are tasks not terminated, the test still can pass, with or without the timeout = Long.MAX_VALUE.

Hmm, this is not expected. The timeout is used in the `SplitFetcherManager.close()`
```
executors.awaitTermination(timeoutMs, TimeUnit.MILLISECONDS)
```
This is a blocking call and won't return until the executors finishes all the tasks, or otherwise be interrupted.

So if the timeout is `Long.MAX_VALUE`, and there are running tasks, `SplitFetcherManager.close()` will block.

Reply:
Sorry @becketqin , I misunderstood your comment. I thought you were asking me to change this:
```
@Test(timeout = Long.MAX_VALUE)
```

My bad! Just read it again, I understand what you said now. We should set the max timeout on the `close` method, otherwise, it will only log a warning and the test passes. So everything makes sense now. Just updated the PR. Please take a look again. Thanks.